### PR TITLE
Linear isotropic hardening

### PR DESCRIPTION
**Pull request by Sina-av**
Hi!
As the code already has some hardening laws, to be complete and easy to use I added the isotropic hardening law.
Based on works of [Yin et al.](https://www.researchgate.net/publication/341423428_A_ductile_phase-field_model_based_on_degrading_the_fracture_toughness_Theory_and_implementation_at_small_strain) and [Ambati et al.](https://link.springer.com/article/10.1007/s00466-015-1151-4) degradation function is not applied to the yield surface. (But it can be done easily)
Please, let me know if you think this is a good idea, or if it needs work.
Thanks.

### CODE REVIEW

**Comment by hugary1995** [comment]
Thank you very much for your contribution! This looks good to me. We didn't include a linear hardening law because we thought it is a special case of `PowerLawHardening`. But maybe their underlying energies are different.

To merge your contribution, all of the tests above have to pass. There are several things you need to do to get the tests pass:
1. All pull request must reference a valid issue number. If none of the existing issues matches your pull request, you can create one. This time, I have created an issue for you #100, so you can directly reference it. To reference an issue, you will need to modify an existing commit message (or create a new commit message) to say something like "ref #100", so that the test will pass.
2. All C++ files in RACCOON must be clang-formatted. If you click on "Details" to the right of "
clang-format / Check clang-format (src) (pull_request)", you can see which of your file doesn't comply with clang-format. After you've corrected the format of that file, you can commit and push it, and that test will pass.

Please don't hesitate to ask if you have any questions. Thanks again.

File: include/materials/hardening_models/LinearIsotropicHardening.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// By: Sina A. V. Based on RACCOON
```
Comment (hugary1995) [inline include/materials/hardening_models/LinearIsotropicHardening.h:1]: You can add your authorship here -- no problem. But the original license heading has to be kept.

File: src/materials/hardening_models/LinearIsotropicHardening.C [hunk 2]
```diff
@@ -0,0 +1,66 @@
+// By: Sina A. V. Based on RACCOON
```
Comment (hugary1995) [inline src/materials/hardening_models/LinearIsotropicHardening.C:1]: You can add your authorship here -- no problem. But the original license heading has to be kept.

**Review by hugary1995 — CHANGES_REQUESTED** [review]
Just two minor requests. Otherwise this looks perfect.

File: include/materials/hardening_models/LinearIsotropicHardening.h (hunk 1 above)
Comment (Sina-av) [reply include/materials/hardening_models/LinearIsotropicHardening.h:1]: Of course, i'll add that I was not sure if I can keep it.

**Review by hugary1995 — APPROVED** [review]

**Comment by hugary1995** [comment]
Okay this looks good to me now. I'll wait for the final test to pass before I merge it. 

I'm expecting the documentation test to fail, but that's fine for now.
- I'll add a documentation page for this new hardening model in a separate PR.
- I'll also add a regression test for this model so that it doesn't break in the future.

@Sina-av Thanks for your contribution.

**Pull request merged**